sdks/python/apache_beam/io/tfrecordio.py

```diff
@@ -0,0 +1,274 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""TFRecord sources and sinks."""
+
+from __future__ import absolute_import
+
+import logging
+import struct
+
+from apache_beam import coders
+from apache_beam.io import filebasedsource
+from apache_beam.io import fileio
+from apache_beam.io.iobase import Read
+from apache_beam.io.iobase import Write
+from apache_beam.transforms import PTransform
+import crcmod
+
+__all__ = ['ReadFromTFRecord', 'WriteToTFRecord']
+
+
+_crc32c_fn = None
+try:
+  import snappy  # pylint: disable=import-error
+  _crc32c_fn = snappy._crc32c  # pylint: disable=protected-access
+except ImportError:
+  logging.warning('Couldn\'t find python-snappy so the implementation of '
```

Comment:
Ah... It was showing the warning when importing the module.
I changed it so that the warning is shown only at the first time when the crc32 is calculated.